src/network/netdev/wireguard.c

```diff
@@ -687,6 +688,16 @@ static void wireguard_done(NetDev *netdev) {
                 }
                 free(peer);
         }
+
+        while ((e = w->unresolved_endpoints)) {
```

Comment:
For this to work, it must be ensured that the timer that resolves endpoints no longer tries to access the data here. I have to recheck if that is the case when `sd_event_source_unref` is applied on it.

Reply:
From `sd_event_source_unref(3)`:

      Note that event source objects stay alive and may be dispatched as long
       as they have a reference counter greater than zero. In order to drop a
       reference of an event source and make sure the associated event source
       handler function is not called anymore it is recommended to combine a
       call of sd_event_source_unref() with a prior call to
       sd_event_source_set_enabled() with SD_EVENT_OFF.

Can you also call `sd_event_source_set_enabled` in this case?

Reply:
In general, if we want to disable an event, actually `sd_event_source_set_enabled()` needs to be called. But, in this case, the `sd_event_source` object is referenced only by `Wireguard`. So, it is not necessary to call the function, that is, `sd_event_source_unref()` is enough here.

We usually do not call `sd_event_source_set_enabled()` in the similar cases.